File: app/src/main/java/com/example/movieapi/api/APIService.java
package com.example.movieapi.api;

import com.example.movieapi.model.MovieDescription;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface APIService {

    @GET("/3/movie/{sorting}/")
    Call<MovieList> getMovieWithId(
            @Path("sorting") String sorting,
            @Query("api_key") String apiKey,
            @Query("language") String language,
            @Query("page") int page
    );

    @GET("/3/movie/{id}")
    Call<Description> getDescriptionWithId(
            @Path("id") String id,
            @Query("api_key") String apiKey,
            @Query("language") String language
    );
}

File: app/src/main/java/com/example/movieapi/api/APIService.kt
package com.example.movieapi.api

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface APIService {
    @GET("/3/movie/{sorting}/")
    fun getMovieWithId(
            @Path("sorting") sorting: String?,
            @Query("api_key") apiKey: String?,
            @Query("language") language: String?,
            @Query("page") page: Int
    ): Call<MovieList?>?

    @GET("/3/movie/{id}")
    fun getDescriptionWithId(
            @Path("id") id: String?,
            @Query("api_key") apiKey: String?,
            @Query("language") language: String?
    ): Call<Description?>?
}